### DISCUSSION TITLE
serialudpi problem

### DISCUSSION THREAD

**Discussion by SimonR49** [avrdudes/avrdude#1696] [Q&A] [answered]
Hi, First attempt to use avrdude fails: windows-x64 build, avrdude says:
I:\Electronics\avrdude-v7.3-windows-x64>avrdude -P ?s -c serialupdi -p avr32ea32
avrdude error: avrdude built without libserialport support; please compile again with libserialport installed

I just want to program the chip - not get into building tools.
Help please?
Thanks

**Comment by MCUdude** [comment, marked as answer]
> I:\Electronics\avrdude-v7.3-windows-x64>avrdude -P ?s -c serialupdi -p avr32ea32

The you'll have to specify the actual COM port, for instance: `I:\Electronics\avrdude-v7.3-windows-x64>avrdude -P COM3 -c serialupdi -p avr32ea32`, as a port search requires a binary build with libserialport. Did you use one of the official binaries?

**Comment by SimonR49** [comment]
It works! device signature found!
Thanks for such a quick response.
Yes, windows build from github.
Regards,
Simon

**Reply by MCUdude** [reply]
Great to hear you got it working! @mcuee maybe we should provide binaries that are built _with_ libserialport?

**Reply by mcuee** [reply]
@MCUdude 

Please refer to this discussion, I have already uploaded the MSYS2 mingw64/mingw32 binaries which supports libserialport. I have re-opened that discussion.
* https://github.com/avrdudes/avrdude/discussions/1675
